extensions/wikia/AdEngine/js/video/player/porvata/floater.js

```diff
@@ -0,0 +1,174 @@
+/*global define, require*/
+define('ext.wikia.adEngine.video.player.porvata.floater', [
+		'wikia.window',
+		'wikia.document'
+	], function (win, doc) {
+		'use strict';
+
+		var activeFloatingCssClass = 'floating',
+			compatibleSlots = ['TOP_LEADERBOARD'],
+			floatingVideoPadding = 28;
+
+		function resetAdContainer(top) {
+			top.style.removeProperty('width');
+			top.style.removeProperty('height');
+		}
+
+		function updateDimensions(element, width, height) {
+			if (element) {
+				element.style.width = width + 'px';
+				element.style.height = height + 'px';
+			}
+		}
+
+		function updateImage(image, width, height) {
+			if (image) {
+				image.width = width;
+				image.style.marginTop = ((height - image.height) / 2) + 'px';
+			}
+		}
+
+		function createCloseButton() {
+			var a = doc.createElement('a');
+
+			a.classList.add('floating-close-button');
+			a.classList.add('hidden');
+
+			return a;
+		}
+
+		function createOnCloseListener(elements, params, floatingContext, imageMarginTop) {
+			return function () {
+				disableFloating(elements, params, imageMarginTop);
+				win.removeEventListener('scroll', floatingContext.scrollListener);
+
+				if (floatingContext.onClose) {
+					floatingContext.onClose();
+				}
+
+				deleteCloseButton(elements, floatingContext);
+			};
+		}
+
+		function deleteCloseButton(elements, floatingContext) {
+			if (floatingContext.closeButton) {
+				elements.top.removeChild(floatingContext.closeButton);
+			}
+		}
+
+		function resetDimensions(element, params) {
+			updateDimensions(element, params.width, params.height);
+		}
+
+		function resetImage(image, marginTop) {
+			if (image) {
+				image.removeAttribute('width');
+				image.style.marginTop = marginTop;
+			}
+		}
+
+		function enableFloating(elements) {
+			var width = ((win.innerWidth - elements.background.offsetWidth) / 2) - floatingVideoPadding,
```

Comment:
I'm not sure how does it really look like, but isn't it too big? Just curious - #ama-nef :P